cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorageClient.java

```diff
@@ -0,0 +1,36 @@
+package com.linkedin.openhouse.cluster.storage;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * BaseStorageClient is the abstract class that holds common functionality for all storage clients
+ * like validating the configuration and storage properties for a given storage type.
+ *
+ * @param <T> native client type for the storage backend.
+ */
+public abstract class BaseStorageClient<T> implements StorageClient<T> {
+  // Validates the configuration for the given storage type. Validates that:
+  // 1. The storageProperties contain the entry for the given storage type.
+  // 2. The storageProperties value for the given storage type is not null and has the endpoint and
+  // root path configured.
+  protected void validateProperties(
+      StorageProperties storageProperties, StorageType.Type storageType) {
```

Comment:
we can leverage `StorageClient::getType()` here, once its added as suggested in the other comment.

Reply:
Good point. Also noticed that the code here has remnants from copy-paste and still has references to hdfsStorageProperties variable name. Fixed that.